app/src/main/java/de/grobox/transportr/settings/SettingsFragment.kt

```diff
@@ -91,6 +103,37 @@ class SettingsFragment : PreferenceFragmentCompat() {
                 true
             }
         }
+
+        arrayOf(PROXY_ENABLE, PROXY_HOST, PROXY_PORT, PROXY_PROTOCOL).forEach { prefKey ->
+            (findPreference(prefKey) as Preference?)?.let { pref ->
+                pref.setOnPreferenceChangeListener { _, newValue ->
+                    try {
+                        val newProxy = settingsManager.getProxy(mapOf(
+                            Pair(prefKey, newValue)
+                        ))
+                        TransportrUtils.checkInternetConnectionViaProxy(newProxy)
+                        TransportrUtils.updateGlobalHttpProxy(newProxy, manager)
+                    } catch (e: Exception) {
+                        Log.e(TAG, "Invalid proxy settings: " + e.message)
+                        AlertDialog.Builder(context!!)
```

Comment:
I would prefer a less disturbing Toast message at this point.

Reply:
Do you think it would be a good idea to handle some common errors using a Toast with a simplified message and use an AlertDialog with the Exception's message for anything unexpected?